cassovary-core/src/main/scala/com/twitter/cassovary/graph/UndirectedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+
+/*
+ * A graph where the outbound neighbors of each node equal the inbound neighbors.
+ * Given a DirectedGraph which stores the neighbors of each node (as outbound edges),
+ * this class wraps that graph in an undirected view.
+ */
+
+class UndirectedGraph(outboundGraph: DirectedGraph) extends DirectedGraph {
```

Comment:
Why does it matter that we return an UndirectedNode, as opposed to a Node which happens to have the same outbound and inbound neighbors?  Relating to Szymon's comment, what properties do undirected nodes have that Nodes don't have?